server/src/test/java/io/crate/metadata/doc/DocSchemaInfoTest.java

```diff
@@ -130,4 +142,100 @@ public void testNoNPEIfDeletedIndicesNotInPreviousClusterState() throws Exceptio
         docSchemaInfo.invalidateFromIndex(new Index("my_index", "asdf"), metadata);
     }
 
+    @Test
+    public void test_update_on_logical_publication_changes() throws Exception {
```

Comment:
Could you turn this into multiple specific/focused test cases?

Makes it easier to maintain this in the long run.